libs/community/langchain_community/chat_models/huggingface.py

```diff
@@ -41,7 +38,7 @@ class ChatHuggingFace(BaseChatModel):
     Adapted from: https://python.langchain.com/docs/integrations/chat/llama2_chat
     """
 
-    llm: Union[HuggingFaceTextGenInference, HuggingFaceEndpoint, HuggingFaceHub]
```

Comment:
this is also backwards incompatible, can we revert?

Reply:
Only problem is when I add this line back, it seems to confuse the deprecation manager:
Now running this script
```
from langchain_community.llms import HuggingFaceEndpoint
from langchain.chat_models import ChatHuggingFace

llm = HuggingFaceEndpoint(repo_id="mistralai/Mixtral-8x7B-Instruct-v0.1")
agent = ChatHuggingFace(llm=llm)
```
Triggers this error:
```
---------------------------------------------------------------------------
NotImplementedError                       Traceback (most recent call last)
[/home/ubuntu/benchmark_agents/test_frameworks.ipynb](https://vscode-remote+ssh-002dremote-002bec2.vscode-resource.vscode-cdn.net/home/ubuntu/benchmark_agents/test_frameworks.ipynb) Cell 1 line 5
      [2](vscode-notebook-cell://ssh-remote%2Bec2/home/ubuntu/benchmark_agents/test_frameworks.ipynb#X56sdnNjb2RlLXJlbW90ZQ%3D%3D?line=1) from langchain.chat_models import ChatHuggingFace
      [4](vscode-notebook-cell://ssh-remote%2Bec2/home/ubuntu/benchmark_agents/test_frameworks.ipynb#X56sdnNjb2RlLXJlbW90ZQ%3D%3D?line=3) llm = HuggingFaceEndpoint(repo_id="mistralai/Mixtral-8x7B-Instruct-v0.1")
----> [5](vscode-notebook-cell://ssh-remote%2Bec2/home/ubuntu/benchmark_agents/test_frameworks.ipynb#X56sdnNjb2RlLXJlbW90ZQ%3D%3D?line=4) agent = ChatHuggingFace(llm=llm)

File [~/langchain_fuse_hf_endpoints/libs/community/langchain_community/chat_models/huggingface.py:50](https://vscode-remote+ssh-002dremote-002bec2.vscode-resource.vscode-cdn.net/home/ubuntu/benchmark_agents/~/langchain_fuse_hf_endpoints/libs/community/langchain_community/chat_models/huggingface.py:50), in ChatHuggingFace.__init__(self, **kwargs)
     [49](https://vscode-remote+ssh-002dremote-002bec2.vscode-resource.vscode-cdn.net/home/ubuntu/benchmark_agents/~/langchain_fuse_hf_endpoints/libs/community/langchain_community/chat_models/huggingface.py:49) def __init__(self, **kwargs: Any):
---> [50](https://vscode-remote+ssh-002dremote-002bec2.vscode-resource.vscode-cdn.net/home/ubuntu/benchmark_agents/~/langchain_fuse_hf_endpoints/libs/community/langchain_community/chat_models/huggingface.py:50)     super().__init__(**kwargs)
     [52](https://vscode-remote+ssh-002dremote-002bec2.vscode-resource.vscode-cdn.net/home/ubuntu/benchmark_agents/~/langchain_fuse_hf_endpoints/libs/community/langchain_community/chat_models/huggingface.py:52)     from transformers import AutoTokenizer
     [54](https://vscode-remote+ssh-002dremote-002bec2.vscode-resource.vscode-cdn.net/home/ubuntu/benchmark_agents/~/langchain_fuse_hf_endpoints/libs/community/langchain_community/chat_models/huggingface.py:54)     self._resolve_model_id()

File [~/venv/agents/lib/python3.10/site-packages/langchain_core/load/serializable.py:107](https://vscode-remote+ssh-002dremote-002bec2.vscode-resource.vscode-cdn.net/home/ubuntu/benchmark_agents/~/venv/agents/lib/python3.10/site-packages/langchain_core/load/serializable.py:107), in Serializable.__init__(self, **kwargs)
    [106](https://vscode-remote+ssh-002dremote-002bec2.vscode-resource.vscode-cdn.net/home/ubuntu/benchmark_agents/~/venv/agents/lib/python3.10/site-packages/langchain_core/load/serializable.py:106) def __init__(self, **kwargs: Any) -> None:
--> [107](https://vscode-remote+ssh-002dremote-002bec2.vscode-resource.vscode-cdn.net/home/ubuntu/benchmark_agents/~/venv/agents/lib/python3.10/site-packages/langchain_core/load/serializable.py:107)     super().__init__(**kwargs)
    [108](https://vscode-remote+ssh-002dremote-002bec2.vscode-resource.vscode-cdn.net/home/ubuntu/benchmark_agents/~/venv/agents/lib/python3.10/site-packages/langchain_core/load/serializable.py:108)     self._lc_kwargs = kwargs

File [~/venv/agents/lib/python3.10/site-packages/pydantic/v1/main.py:339](https://vscode-remote+ssh-002dremote-002bec2.vscode-resource.vscode-cdn.net/home/ubuntu/benchmark_agents/~/venv/agents/lib/python3.10/site-packages/pydantic/v1/main.py:339), in BaseModel.__init__(__pydantic_self__, **data)
    [333](https://vscode-remote+ssh-002dremote-002bec2.vscode-resource.vscode-cdn.net/home/ubuntu/benchmark_agents/~/venv/agents/lib/python3.10/site-packages/pydantic/v1/main.py:333) """
    [334](https://vscode-remote+ssh-002dremote-002bec2.vscode-resource.vscode-cdn.net/home/ubuntu/benchmark_agents/~/venv/agents/lib/python3.10/site-packages/pydantic/v1/main.py:334) Create a new model by parsing and validating input data from keyword arguments.
    [335](https://vscode-remote+ssh-002dremote-002bec2.vscode-resource.vscode-cdn.net/home/ubuntu/benchmark_agents/~/venv/agents/lib/python3.10/site-packages/pydantic/v1/main.py:335) 
    [336](https://vscode-remote+ssh-002dremote-002bec2.vscode-resource.vscode-cdn.net/home/ubuntu/benchmark_agents/~/venv/agents/lib/python3.10/site-packages/pydantic/v1/main.py:336) Raises ValidationError if the input data cannot be parsed to form a valid model.
    [337](https://vscode-remote+ssh-002dremote-002bec2.vscode-resource.vscode-cdn.net/home/ubuntu/benchmark_agents/~/venv/agents/lib/python3.10/site-packages/pydantic/v1/main.py:337) """
    [338](https://vscode-remote+ssh-002dremote-002bec2.vscode-resource.vscode-cdn.net/home/ubuntu/benchmark_agents/~/venv/agents/lib/python3.10/site-packages/pydantic/v1/main.py:338) # Uses something other than `self` the first arg to allow "self" as a settable attribute
...
    [339](https://vscode-remote+ssh-002dremote-002bec2.vscode-resource.vscode-cdn.net/home/ubuntu/benchmark_agents/~/venv/agents/lib/python3.10/site-packages/langchain_core/_api/deprecation.py:339)     )
    [340](https://vscode-remote+ssh-002dremote-002bec2.vscode-resource.vscode-cdn.net/home/ubuntu/benchmark_agents/~/venv/agents/lib/python3.10/site-packages/langchain_core/_api/deprecation.py:340) else:
    [341](https://vscode-remote+ssh-002dremote-002bec2.vscode-resource.vscode-cdn.net/home/ubuntu/benchmark_agents/~/venv/agents/lib/python3.10/site-packages/langchain_core/_api/deprecation.py:341)     removal = f"in {removal}"

NotImplementedError: Need to determine which default deprecation schedule to use. within ?? minor releases
```

Reply:
The error seems to be triggered [here](https://github.com/langchain-ai/langchain/blob/53b8c863099df926215946c06be102d199c64d21/libs/core/langchain_core/_api/deprecation.py#L336) and seems due to the fact that I have not setup neither `pending` nor `removal` argument.
@baskaryan : do you know the proper configuration I should use in my `@deprecated` decorator?

Reply:
ah yea let's add a removal argument, can set `removal=0.2.0`